File: Problems/0455-assign-cookies.py
class Solution(object):
    def findContentChildren(self, g, s):
        g.sort()
        s.sort()

        i = j = 0
        m = len(g)
        n = len(s)

        while i < m and j < n:
            if s[j] >= g[i]:
                i += 1
            j += 1

        return i

File: Problems/0455-assign-cookies.cpp
#include <bits/stdc++.h>
using namespace std;

// LeetCode solution starts from here
class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        sort(g.begin(), g.end());
        sort(s.begin(), s.end());

        int i = 0, j = 0;
        int m = g.size(), n = s.size();

        while (i < m && j < n) {
            if (s[j] >= g[i])
                i++;
            j++;
        }

        return i;
    }
};